index/segment/mem/terms_dict.go

```diff
@@ -53,19 +53,37 @@ func (d *termsDict) Insert(field doc.Field, id postings.ID) error {
 	return postingsMap.Add(field.Value, id)
 }
 
+func (d *termsDict) ContainsTerm(field, term []byte) (bool, error) {
+	_, found, err := d.matchTerm(field, term)
```

Comment:
This still allocates if not found. could you re-work to avoid that.

Reply:
I'm not sure I see how it still allocates if not found?

Reply:
you're right. `matchTerm != MatchTerm` :|